Automattic/vip-design-system
# PIE-3053 - Add async select component

## Description

Adding an async version of our `SearchSelect` component to allow for paginated results to load asynchronously both in the dropdown selector list and via search.

This allows you to search for options that aren't loaded fully (i.e.- site 115, which is beyond the default 100 queries). You can also scroll to the bottom of the dropdown list to query the next set of paginated results asynchronously:
https://user-images.githubusercontent.com/29221840/159411417-f9beb5c7-5332-4f10-8fea-d8a10489cc49.mov

## Checklist


## Steps to Test

1. Pull down PR.
1. `npm run storybook`
1. Navigate to the component on Storybook (Storybook > Select> Async): http://localhost:6006/?path=/story/select--async
1. Dropdown selection + search should work*

_*This is a very simple rendition of the component. Ideally, you can link to the Dashboard to see it in action (see video above)._

5. Run `npm run test`
6. All tests should pass*

P.S: There's currently a lot of errors due to a semi-related component. Will open a separate PR to address those errors:

```
Warning: Invalid value for prop `cx` on <svg> tag. Either remove it from the element, or pass a string or number value to keep it in the DOM.
```
